pandas/core/internals/construction.py

```diff
@@ -152,9 +152,17 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
         return arrays_to_mgr([values], columns, index, columns, dtype=dtype)
     elif is_extension_array_dtype(values) or is_extension_array_dtype(dtype):
         # GH#19157
+
+        if isinstance(values, np.ndarray) and values.ndim > 1:
```

Comment:
actually this is likely duplicating the category checks above

Reply:
need to handle the copy arg

Reply:
> actually this is likely duplicating the category checks above

Yah we can probably do a grep for is_categorical_dtype and find a lot of things to clean up (adding this to my grep list...)

> need to handle the copy arg

how so?  copy is passed to prep_ndarray below, and so far all we've done is take views